packages/tabs/src/useTabs.ts

```diff
@@ -9,16 +9,46 @@ import { useState } from 'react';
 import { useSelection } from '@zendeskgarden/container-selection';
 import { generateId } from '@zendeskgarden/container-utilities';
 
-function requiredArguments(arg, argStr, methodName) {
+interface IUseTabsPropGetters {
+  getTabProps: <T>(options?: T & IGetTabPropsOptions) => T & IGetTabPropsOptions;
```

Comment:
`getTabProps` doesn't actually return `index` from it's prop-getter (`IGetTabPropsOptions`) so this definition would fail if spread onto an element (index isn't a valid prop).

This also extends `useSelection` so we should be using those interfaces as the base.

Take a look at `useButtonGroup` for a valid example

```tsx
export interface IUseButtonGroupPropGetters<Item> {
  getGroupProps: <T>(options?: T) => T & React.HTMLProps<any>;
  getButtonProps: <T extends IGetItemPropsOptions<Item>>(
    options: T,
    propGetterName?: string
  ) => any;
}
```